docs/docs/code/python/delay/README.md

```diff
@@ -0,0 +1,122 @@
+
+    # Delay a workflow a random amount of time
+    pd.flow.delay(random.randint(0, 999))
+```
+
+::: tip Paused workflow state
+
+When `pd.flow.delay` is executed in a Python step, the workflow itself will enter a **Paused** state.
+
+While the workflow is paused, it will not incur any credits towards compute time. You can also [view all paused workflows in the Event History](/event-history/#filtering-by-status).
+
+:::
+
+### Credit usage
+
+The length of time a workflow is delayed from `pd.flow.delay` does _not_ impact your credit usage. For example, delaying a 256 megabyte workflow for five minutes will **not** incur ten credits.
+
+However, using `pd.flow.delay` in a workflow will incur two credits.
+
+One credit is used to initially start the workflow, then the second credit is used when the workflow resumes after its pause period has ended.
+
+## `cancel_url` and `resume_url`
+
+Both the built-in **Delay** actions and `pd.flow.delay` return a `cancel_url` and `resume_url` that lets you cancel or resume paused executions.
+
+These URLs are specific to a single execution of your workflow. While the workflow is paused, you can load these in your browser or send an HTTP request to either:
+
+- Hitting the `cancel_url` will immediately cancel that execution
+- Hitting the `resume_url` will immediately resume that execution early
+
+[Since Pipedream pauses your workflow at the _end_ of the step where you run call `pd.flow.delay`](#when-delays-happen), you can send these URLs to third party systems, via email, or anywhere else you'd like to control the execution of your workflow.
+
+```python
+import requests
+
+def handler(pd: 'pipedream'):
+  links = pd.flow.delay(15 * 60 * 1000)
+  # links contains a dictionary with two entries: resume_url and cancel_url
+
+  # Send the URLs to a system you own
+  requests.post("https://example.com", json=links)
+
+  # Email yourself the URLs. Click on the links to cancel / resume
+  pd.send.email(
+    subject=f"Workflow execution {pd.steps['trigger']['context']['id']}",
+    text=f"Cancel your workflow here: {links['cancel_url']} . Resume early here: {links['resume_url']}",
+    html=None
+  )
+
+  # Delay happens at the end of this step
+```
+
+## When delays happen
+
+**Pipedream pauses your workflow at the _end_ of the step where you call `pd.flow.delay`**. This lets you [send the `cancel_url` and `resume_url` to third-party systems](#cancel-url-and-resume-url).
+
+```python
+def handler(pd: 'pipedream'):
+  urls = pd.flow.delay(15 * 60 * 1000)
+  cancel_url, resume_url = urls["cancel_url"], urls["resume_url"]
+  # ... run any code you want here
+
+  # Delay happens at the end of this step
+```
+
+## Delays and HTTP responses
+
+You cannot run `pd.respond` after running `pd.flow.delay`. Pipedream ends the original execution of the workflow when `pd.flow.delay` is called and issues the following response to the client to indicate this state:
+
+> $.respond() not called for this invocation
```

Comment:
Is the error emitted in Python still use `$.respond`? Or does it say `pd.respond() not called for this invocation`?

Reply:
Yes, it still says `$.respond() not called for this invocation`.

Reply:
Ah ok, minor DX issue then. Just curious.